### ISSUE TITLE
Kubeflow mode doesn't work for AKS 

### ISSUE THREAD

**Issue by MingzhouZhou** [microsoft/nni#1084]
**Short summary about the issue/question**:
Hi, I have a small AKS cluster and have Kubeflow deployed. I can launch and run distributed-mnist in the AKS-kubeflow cluster from my centOS host. 

**Brief what process you are following**: 
I try to run NNI's mnist-distributed example in the AKS-kubeflow cluster, and I followed the KubeflowMode.md (https://github.com/microsoft/nni/blob/master/docs/en_US/KubeflowMode.md) to set up a vault-managed azure storage account, and verified by powershell command `Get-AzureKeyVaultManagedStorageAccount`. However, the execution of `nnictl create --config config_kubeflow.yml --port 8888` failed very quickly, with error as shown below(I changed nnictl code a bit to get more error info):
```
> $ nnictl create --config config_kubeflow.yml --port 8888
INFO: expand searchSpacePath: search_space.json to /path/nni/examples/trials/mnist-distributed/search_space.json
INFO: expand codeDir: . to /path/nni/examples/trials/mnist-distributed/.
INFO: Starting restful server...
INFO: Successfully started Restful server!
INFO: Setting kubeflow config...
INFO: cluster_metadata_url: http://localhost:8888/api/v1/nni/experiment/cluster-metadata
ERROR: response code: 500
ERROR: Failed! Error is: {"error":"Forbidden"}
```
Looks like the local `Kubeflow training service` fails. The only difference I can see is that I'm on a centOS host, not sure if this matters. Any thoughts? Thanks in advance! 

**How to reproduce it**: 

**nni Environment**:
- nni version: v0.7-18-gaf89df8
- nni mode(local|pai|remote): kubeflow
- OS: centOS
- python version: 3.6.1
- is conda or virtualenv used?: no
- is running in docker?: no

**need to update document(yes/no)**:

**Anything else we need to know**:

**Comment by SparkSnail** [comment]
Hi, this issue may be caused by the environment permission, let's check the environment first. NNI use KeyVault service to store storage account and connectionKey, you should store the key in keyVault service first, and you could check from the command `az keyvault secret show --name ${valutKeyName} --vault-name ${vaultName}`, refer the code https://github.com/microsoft/nni/blob/master/src/nni_manager/training_service/kubernetes/kubernetesTrainingService.ts#L138.

**Comment by MingzhouZhou** [comment]
Hi @SparkSnail, thanks for the reply. I'm able to run `az keyvault secret show --name ${valutKeyName} --vault-name ${vaultName}`, and tried `nnictl create --config config_kubeflow.yml` again, got the same error. The error seems indicating the local `Kubeflow training service` isn't ready to accept request as it fails at `INFO: Setting kubeflow config...`, I can pinpoint the failure to code here: https://github.com/microsoft/nni/blob/master/tools/nni_cmd/launcher.py#L244, with response code 500. However, the stdout tells `INFO: Successfully started Restful server!`. Any feedback would be highly appreciated. Thanks!

**Comment by MingzhouZhou** [comment]
Hi @SparkSnail, I think my issue might be related to the key-vault setup. I'm new to Azure, I followed this blog http://www.wahidsaleemi.com/2017/08/azure-storage-account-keys-automatic-rotation/ to set up a key-vault managed storage account, whose key1 is added to key-vault as a secret. Is it correct? 

BTW, is there any az command that can test storage account via key-vault secret? Thanks!

**Comment by MingzhouZhou** [comment]
Hi @SparkSnail, here is some progress. I re-add the key1 using `az keyvault secret set --vault-name ${vaultName} --name ${secretName}” --value ${key1Value}`, and now I'm able to run the `nnictl create --config config_kubeflow.yml --port 8888`. Although there is new error related to undefined eth0: `Error: getIPV4Address() failed because os.networkInterfaces().eth0 is undefined.`, set `nniManagerIp` help resolve the issue. 

Basically the root cause seems that the secret is somehow not setup properly in key-vault. You may close the issue now. Many thanks!

**Comment by scarlett2018** [comment]
Thanks @MingzhouZhou for the updates, I'm closing the issue per your latest feedback.
